knowledge_base/errors/web_server.rst

```diff
@@ -0,0 +1,43 @@
+Web Server
+==========
+
+502 - Bad Gateway
+-----------------
+
+Error
+^^^^^
+
+HTTP requests fail with status code 504.
+
+.. figure:: ../img/504_timeout.png
+
+    504 returned by *Nginx*
+
+
+Solution
+^^^^^^^^
+
+First, try to figure out what that request takes that long and try to make it faster.
```

Comment:
You set the header to 502 bad gateway but you are describing a 504 gateway timeout here.